### PR TITLE
Add Spinner component

### LINKED ISSUE #15: Spinner
## Overview
A spinner component that can be used as a loading placeholder.

## Use Cases
- Placeholder during async operations
The spinner is an example of an indeterminate progress indicator. They are useful for visualizing an unspecified wait time

## Prior Art/Examples
- [shoelace](https://shoelace.style/components/spinner)
- [material ui](https://material-ui.com/components/progress/#progress)
- [fast design](https://www.fast.design/docs/components/progress) the progress ring section
- [base web](https://baseweb.design/components/spinner/#spinner)
- [theme ui](https://theme-ui.com/components/spinner)
- [chakra ui](https://chakra-ui.com/docs/feedback/spinner)

---

**Note: The following section is optional for making the issue, but must be filled in for a possible PR to be merged**

## API
*Component name:*
* `<fl-spinner></fl-spinner>`

*Attributes:*
* N/A

*Slots:*
* N/A

*CSS Parts:*
* N/A

## Example usage
```
<fl-spinner></fl-spinner>
```

## Questions to be answered
- Should styling the spinner (size, stroke width, color) be exposed through CSS? Would it be better to expose attributes that set those styles instead?

### PR DESCRIPTION
**Pull request by its-edmund**

### CODE REVIEW

File: packages/core/src/components/spinner/readme.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+title: spinner
+---
+
+# fl-spinner
+
+This is a spinner
+
+## Examples
+
+<fl-spinner size='md'></fl-spinner>
```
Comment (1ntEgr8) [inline packages/core/src/components/spinner/readme.md]: `medium` is more clear. Can we switch it to that?

File: packages/core/src/components/spinner/spinner.scss [hunk 2]
```diff
@@ -0,0 +1,54 @@
+:host {
+  --p: 0;
```
Comment (1ntEgr8) [inline packages/core/src/components/spinner/spinner.scss]: What does this variable do?

Also, just an idea (would probably be best as a separate PR), we could use [CSS parts](https://developer.mozilla.org/en-US/docs/Web/CSS/::part) to customize the spinner.

File: packages/core/src/components/spinner/spinner.scss [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  --q: 0.7;
```
Comment (1ntEgr8) [inline packages/core/src/components/spinner/spinner.scss]: ditto ^

File: packages/core/src/components/spinner/spinner.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { Component, h, Host, Prop } from "@stencil/core";
+
+@Component({
+  tag: "fl-spinner",
+  styleUrl: "spinner.scss",
+  shadow: true,
+})
+export class Spinner {
+  @Prop() size: "sm" | "md" | "lg" = "md";
```
Comment (1ntEgr8) [inline packages/core/src/components/spinner/spinner.tsx]: I think we should change these to `small`, `medium`, and `large` because the purpose is more clear

File: packages/core/src/components/spinner/spinner.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+  render() {
+    return (
+      <Host>
```
Comment (1ntEgr8) [inline packages/core/src/components/spinner/spinner.tsx:13]: Might be missing something, but we don't need to use `Host` here right?

**Review by 1ntEgr8 — APPROVED** [review]
lgtm after addressing comments!

File: packages/core/src/components/spinner/spinner.tsx (hunk 5 above)
Comment (its-edmund) [reply packages/core/src/components/spinner/spinner.tsx:13]: Makes setting the props a lot easier for some reason.

**Comment by its-edmund** [comment]
I'll keep the branch in case we want to implement css parts
